chains/ethereum/header_store.go

```diff
@@ -65,17 +65,39 @@ func headerKey(number uint64, hash common.Hash) string {
 	return fmt.Sprintf("%d%s%s", number, SplicingSymbol, hash.Hex())
 }
 
+func (hs *HeaderStore) headerIdx(number uint64) uint64 {
+	return uint64(math.Mod(float64(number), MaxHeaderLimit)) + startNumber
```

Comment:
startNumber can not be initialized when reboot the node.

Reply:
Yes, we have made improvements here. We do not use startNumber